### PR TITLE
Comment screen - Add comment

### PR DESCRIPTION
**Pull request by at-huyvo2**
- Add comment to photo

### CODE REVIEW

File: MyApp/View/Controller/CommentViewController/CommentViewController.swift [hunk 1]
```diff
@@ -101,7 +115,11 @@ final class CommentViewController: BaseViewController {
     }
 
     @IBAction func sendButtonTouchUpInside(_ sender: Any) {
-        addComment()
+        guard let comment = commentTextView.text else { return }
+        if !comment.isEmpty {
```
Comment (at-hainguyen3) [inline MyApp/View/Controller/CommentViewController/CommentViewController.swift]: Đặt điều kiện lên trong guard let luôn, không cần if nữa

File: MyApp/View/Controller/CommentViewController/CommentViewController.swift [hunk 2]
```diff
@@ -146,5 +167,6 @@ extension CommentViewController {
         static let hideKeyboardTabbarHieght: CGFloat = 0
         static let hideKeyboardAddButtonWidth: CGFloat = 0
         static let showKeyboardAddButtonWidth: CGFloat = 46
+        static let placeholderText = "Add your comment"
```
Comment (at-hainguyen3) [inline MyApp/View/Controller/CommentViewController/CommentViewController.swift]: Define text

File: MyApp/View/Controller/DetailViewController/DetailViewController.swift [hunk 3]
```diff
@@ -221,6 +221,11 @@ final class DetailViewController: BaseViewController {
         }
         present(ImageViewerController(configuration: configuration), animated: true)
     }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        callApis()
```
Comment (at-hainguyen3) [inline MyApp/View/Controller/DetailViewController/DetailViewController.swift]: Vì sao lại call api ở đây

Comment (at-huyvo2) [reply MyApp/View/Controller/DetailViewController/DetailViewController.swift]: để pop view nó hiện cái cmt mình mới add anh

File: MyApp/Library/Ext/UIScrollExt.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+extension UIScrollView {
+
+    private(set) var lastContentOffsetY: CGFloat = 0
+
+    enum ScrollDirection {
+        case up
+        case down
+        case none
+    }
+
```
Comment (at-daiho) [inline MyApp/Library/Ext/UIScrollExt.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/Library/Ext/UIScrollExt.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        case none
+    }
+    
+    var scrollDirection: ScrollDirection {
+        guard lastContentOffsetY = lastContentOffsetY else {
+            lastContentOffsetY = 0
+            return
+        }
+
+
```
Comment (at-daiho) [inline MyApp/Library/Ext/UIScrollExt.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/Library/Ext/UIScrollExt.swift [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    
+    var scrollDirection: ScrollDirection {
+        guard lastContentOffsetY = lastContentOffsetY else {
+            lastContentOffsetY = 0
+            return
+        }
+
+        
+
```
Comment (at-daiho) [inline MyApp/Library/Ext/UIScrollExt.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/Library/Ext/UIScrollExt.swift [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+            lastContentOffsetY = 0
+            return
+        }
+
+        
+        
+        lastContentOffsetY = self.contentOffset.y
+        return  .up
+    }
+
```
Comment (at-daiho) [inline MyApp/Library/Ext/UIScrollExt.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/Library/Ext/UIScrollExt.swift [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+            return
+        }
+
+        
+        
+        lastContentOffsetY = self.contentOffset.y
+        return  .up
+    }
+    
+
```
Comment (at-daiho) [inline MyApp/Library/Ext/UIScrollExt.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/Library/Ext/UIScrollExt.swift (hunk 6 above)
Comment (at-daiho) [inline MyApp/Library/Ext/UIScrollExt.swift]: WARNING - Limit vertical whitespace to a single empty line. Currently 3.

File: MyApp/Library/Ext/UIScrollExt.swift (hunk 8 above)
Comment (at-daiho) [inline MyApp/Library/Ext/UIScrollExt.swift]: WARNING - Limit vertical whitespace to a single empty line. Currently 2.

**Comment by at-daiho** [comment]
Total linter violations in pull request: 7
Serious: 0
Warnings: 7

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

**Pull request merged**